internal/abci/abci.go

```diff
@@ -1067,3 +1297,22 @@ func (a *AbciApp) TxInMempool(txHash []byte) bool {
 	_, ok := a.txCache[string(txHash)]
 	return ok
 }
+
+// SetReplayStatusChecker sets the function to check if the node is in replay mode.
+// This has to be set here because it is a CometBFT node function. Since ABCI is
+// a dependency to CometBFT, this is a circular dependency, so we have to set it
+// here.
+func (a *AbciApp) SetReplayStatusChecker(fn func() bool) {
+	a.replayingBlocks = fn
+}
+
+// Close cleans up the application.
+func (a *AbciApp) Close() error {
+	if a.genesisTx != nil {
+		err := a.genesisTx.Rollback(context.Background())
+		if err != nil {
+			return fmt.Errorf("failed to rollback genesis transaction: %w", err)
+		}
+	}
+	return nil
```

Comment:
Here's the removed `Close` from txapp: https://github.com/kwilteam/kwil-db/pull/831/files#diff-29b2bc6069630247b4e88988ee7f8c5a0c4fa090867fc9d1e5d050c64da892d1L129-L138

We'll need to check for both tx fields to avoid a hang in all cases, not just genesis, I believe.  Can probably just grab that old method, docs and all.

Reply:
I think this still needs resolving.  Everything else looks good now.